### ISSUE TITLE
ENH: Read geodatabase that contains raster

### ISSUE THREAD

**Issue by tqa236** [geopandas/geopandas#2793]
#### Is your feature request related to a problem?

I tried to open a geodatabase file that contains a raster file, but the result's geometry is a Polygon with value 1 everywhere. Will this be a new feature or there's already a way to do that?

#### Describe the solution you'd like

It's possible to open and parse correctly a geodatabase that contains a raster file.

#### API breaking implications

[this should provide a description of how this feature will affect the API]

#### Describe alternatives you've considered

I have tried `xarray` or `rasterio` or directly with `Fiona` without success. 

#### Additional context

The data is here: https://www.fs.usda.gov/rds/archive/catalog/RDS-2022-0026, file [RDS-2022-0026_Data_NCNM.zip](https://s3-us-west-2.amazonaws.com/fs.usda.rds/RDS-2022-0026/RDS-2022-0026_Data_NCNM.zip)

```
gpd.read_file(
    "RDS-2022-0026_Data_NCNM/Data/NCNM/NCNM_FSim_BurnProbability.gdb",
    driver="OpenFileGDB",
    layer="FSim_iBP_CC",
)
```

Thank you very much!

**Comment by brendan-ward** [comment]
@tqa236 what do you expect to get back from reading a raster?  GeoPandas is a vector (points, lines, polygons) tool; if you are actually wanting to work with raster data as a raster (even if it were supported by underlying libraries), this is not the right tool.

The reason you get back polygons is because the layer you are reading is a feature class with polygon data, not a raster; that part is working as expected.  I was able to successfully read that vector layer as well, so this isn't an issue with not being able to extract vector data from a geodatabase that contains a mix of rasters and vectors.

To my knowledge, GDAL (used by Fiona, Pyogrio, Rasterio) does not yet support reading geodatabase rasters.  Unfortunately I think you are going to need to use a separate tool, likely ArcGIS, to convert these rasters to a format that GDAL supports, such as GeoTIFF.  From there, you need to use rasterio or xarray to work with the data.

Closing as this is out of scope for GeoPandas.

**Comment by tqa236** [comment]
@brendan-ward thanks for your explanation. I'll try to look for other tools

**Comment by AlexanderJuestel** [comment]
Hello, 
According to https://github.com/OSGeo/gdal/pull/7389, the functionality to read rasters was added to GDAL. It was also implemented in QGIS 3.32 as it seems (https://github.com/qgis/QGIS/pull/52214). 

But I guess rasterio would be the right place to implement that, right?

**Comment by martinfleis** [comment]
@AlexanderJuestel Yes, rasterio is the right place.
